lib/encoding.neon

```diff
@@ -0,0 +1,132 @@
+%|
+ |  File: encoding
+ |
+ |  Functions for encoding and decoding ASCII or Binary data in various formats.
+ |%
+IMPORT bitwise
+
+EXPORT base64Encode
+EXPORT base64Decode
+
+TYPE State IS ENUM
+    byte1
+    byte2
+    byte3
+    byte4
+END ENUM
+
+CONSTANT Base64EncodeTable: String := "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"
+
+LET Base64DecodeTable: Array<Number> := [
+     -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1,  -1, -1,  -1,  -1,  -1,  -1,  -1,    % 0x00 - 0x0f
```

Comment:
There's inconsistent spacing in this table. Column 11 has one fewer leading space than the rest of the columns.